File: blind75/python/001.py
# Trick: Use a Dict. Only need to iterate once
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        values = {}
        for idx, n in enumerate(nums):
            if (target - n) in values:
                return [values[target - n], idx]
            values[n] = idx

File: blind75/cpp/001.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // hash map
        unordered_map<int, int> seenNums;
        for (int idx = 0; idx < nums.size(); idx++)
        {
            // check if element is in hash map
            if (seenNums.find(target - nums[idx]) != seenNums.end())
            {
                return {seenNums[target - nums[idx]], idx};
            }
            else 
            {
                seenNums[nums[idx]] = idx;
            }
                
            
        }
        return {0, 0};
    }
};
